Gems/ROS2/Code/Source/FactorySimulation/ConveyorBeltComponentKinematic.cpp

```diff
@@ -0,0 +1,306 @@
+            }
+        }
+        return {};
+    }
+
+    void ConveyorBeltComponentKinematic::Reflect(AZ::ReflectContext* context)
+    {
+        if (AZ::SerializeContext* serialize = azrtti_cast<AZ::SerializeContext*>(context))
+        {
+            serialize->Class<ConveyorBeltComponentKinematic>()
+                ->Version(1)
+                ->Field("BeltEntityId", &ConveyorBeltComponentKinematic::m_ConveyorEntityId)
+                ->Field("Speed", &ConveyorBeltComponentKinematic::m_speed)
+                ->Field("BeltWidth", &ConveyorBeltComponentKinematic::m_beltWidth)
+                ->Field("SegmentLength", &ConveyorBeltComponentKinematic::m_segmentLength)
+                ->Field("TextureScale", &ConveyorBeltComponentKinematic::m_textureScale)
+                ->Field("MaterialAsset", &ConveyorBeltComponentKinematic::m_materialAsset);
+            if (AZ::EditContext* ec = serialize->GetEditContext())
+            {
+                ec->Class<ConveyorBeltComponentKinematic>("Conveyor Belt Component Kinematic", "Conveyor Belt Component")
+                    ->ClassElement(AZ::Edit::ClassElements::EditorData, "")
+                    ->Attribute(AZ::Edit::Attributes::Category, "ROS2")
+                    ->Attribute(AZ::Edit::Attributes::AppearsInAddComponentMenu, AZ_CRC_CE("Game"))
+                    ->DataElement(
+                        AZ::Edit::UIHandlers::EntityId,
+                        &ConveyorBeltComponentKinematic::m_ConveyorEntityId,
+                        "Conveyor Belt Entity",
+                        "Entity of the conveyor belt")
+                    ->DataElement(
+                        AZ::Edit::UIHandlers::Default, &ConveyorBeltComponentKinematic::m_speed, "Speed", "Speed of the conveyor belt")
+                    ->DataElement(AZ::Edit::UIHandlers::Default, &ConveyorBeltComponentKinematic::m_beltWidth, "Width", "Belt width")
+                    ->DataElement(
+                        AZ::Edit::UIHandlers::Default,
+                        &ConveyorBeltComponentKinematic::m_segmentLength,
+                        "Segment Length",
+                        "Length of simulated segments. Short segments causes larger number individual bodies to simulate and challenges "
+                        "physics engine.")
+                    ->DataElement(
+                        AZ::Edit::UIHandlers::Default,
+                        &ConveyorBeltComponentKinematic::m_textureScale,
+                        "Texture scale",
+                        "Scale of the texture on conveyor belt.")
+                    ->DataElement(
+                        AZ::Edit::UIHandlers::Default,
+                        &ConveyorBeltComponentKinematic::m_materialAsset,
+                        "Material",
+                        "Material asset of the conveyor belt")
+                    ->Attribute(AZ::Edit::Attributes::DefaultAsset, &GetDefaultPhysicsMaterialAssetId)
```

Comment:
Is there benefit to using another material asset as default, knowing we are dealing with conveyor belt behavior?

Reply:
I don understand the question. I think it is beneficial to change the material of the simulated belt.